R/report.R

```diff
@@ -59,13 +64,37 @@ revdep_report_section <- function(title, rows, file) {
   cat_kable(rows, file = file)
 }
 
-#' `revdep_report_problems()` writes the contents of the `problems.md`
-#' file to the screen (corresponding to the current state of the checks).
+#' `revdep_report_problems()` generates a report about packages with check
+#' problems.
 #'
 #' @export
 #' @rdname revdep_report_summary
 
-revdep_report_problems <- function(pkg = ".", file = "", all = FALSE) {
+revdep_report_problems <- function(pkg = ".", file = "", all = FALSE, results = NULL) {
+  ## We show the packages that
+  ## 1. are newly broken
+  ## 2. still broken, if all == TRUE
+  problem <- function(x) {
+    any(x$cmp$change == 1) || (all && any(x$cmp$change == 0))
+  }
+  revdep_report_if(pkg = pkg, file = file, predicate = problem, results = results)
+}
+
+#' `revdep_report_failures()` generates a report about packages that failed
+#' to check (i.e. couldn't install, or timed out).
+#'
+#' @export
+#' @rdname revdep_report_summary
+
+revdep_report_failures <- function(pkg = ".", file = "", results = NULL) {
+  problem <- function(x) {
+    !x$status %in% c("+", "-")
```

Comment:
So we want to show everything here? Even if (say) the new check ran without errors?

Reply:
Isn't that `+`? This should include all non-check failures. It did what I expected in my tests, but I don't know what values status can take.

Reply:
`t+` means that the new check was good, but the old timed out
`i+` means that the new check was good, but the old failed to install
`t-` means that at least one timed out, and new check was bad
`i-` means that at least one failed to install and new check was bad

Reply:
Ok, I think I'm primarily interested `t-` and `i-`. What if both checks timed out

Reply:
`t-` is both timed out.

Reply:
basically a `-` means that we could not check with the new version.